cleanlab_studio/internal/enrichment_utils.py

```diff
@@ -37,83 +43,97 @@ def extract_df_subset(
     return subset_df
 
 
-def get_compiled_regex_list(
-    regex: Union[str, re.Pattern[str], List[re.Pattern[str]]]
-) -> List[re.Pattern[str]]:
-    """Compile the regex pattern(s) provided and return a list of compiled regex patterns."""
+def get_regex_match_or_replacement(
```

Comment:
Just confirming, only a single pattern is used for matching (re.search) but a list of patterns can be used for replacements (re.sub). Previously, this was build that a list of patterns could be used for matching

Reply:
That is correct, the options now are:
- a string: matching/extraction (only one string pattern is allowed)
- a tuple: replacement
- a list of tuples: a series of replacements sequentially